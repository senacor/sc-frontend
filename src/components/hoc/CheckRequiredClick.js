import React from 'react';
import { SimpleErrorDialog } from '../pr/SimpleErrorDialog';
import PropTypes from 'prop-types';

export class CheckRequiredClick extends React.Component {
  state = {
    open: false
  };

  handleOpen = event => {
    this.setState({ open: true });
  };

  handleClose = () => {
    this.setState({ open: false });
  };

  handleClick = event => {
    if (this.props.check()) {
      this.props.onClick(event);
    } else {
      if (!event.disabled) {
        this.handleOpen(event);
      }
    }
  };

  render() {
    return (
      <div>
        {React.cloneElement(this.props.children, {
          releaseButtonClick: this.handleClick
        })}
        <SimpleErrorDialog
          onClose={this.handleClose}
          open={this.state.open}
          message={this.props.message}
        />
      </div>
    );
  }
}

CheckRequiredClick.propTypes = {
  onClick: PropTypes.func.isRequired,
  check: PropTypes.func.isRequired,
  message: PropTypes.string.isRequired
};
