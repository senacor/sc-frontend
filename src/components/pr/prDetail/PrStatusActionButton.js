import React from 'react';
import Button from '@material-ui/core/Button';
import { withStyles } from '@material-ui/core/styles/index';

const styles = theme => ({
  buttonDesktop: {
    position: 'relative',
    backgroundColor: theme.palette.primary['400'],
    color: '#FFF',
    marginBottom: '2%'
  },
  sizeSmall: {}
});

function PrStatusActionButton(props) {
  let { classes, label } = props;
  return (
    <Button
      classes={{ sizeSmall: classes.sizeSmall }}
      size="small"
      className={classes.buttonDesktop}
      onClick={props.releaseButtonClick}
    >
      {label}
    </Button>
  );
}

export default withStyles(styles)(PrStatusActionButton);
