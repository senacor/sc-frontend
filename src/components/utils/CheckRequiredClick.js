import React, { useState } from 'react';
import PropTypes from 'prop-types';

import SimpleErrorDialog from '../pr/SimpleErrorDialog';

export const CheckRequiredClick = ({
  check,
  onClick,
  inputClass,
  children,
  message
}) => {
  const [open, setOpen] = useState(false);

  const handleOpen = event => {
    setOpen(true);
  };

  const handleClose = () => {
    setOpen(false);
  };

  const handleClick = event => {
    if (check()) {
      onClick(event);
    } else {
      if (!event.disabled) {
        handleOpen(event);
      }
    }
  };

  return (
    <div className={inputClass}>
      {React.cloneElement(children, {
        releaseButtonClick: handleClick
      })}
      <SimpleErrorDialog onClose={handleClose} open={open} message={message} />
    </div>
  );
};

CheckRequiredClick.propTypes = {
  onClick: PropTypes.func.isRequired,
  check: PropTypes.func.isRequired,
  message: PropTypes.string.isRequired
};
