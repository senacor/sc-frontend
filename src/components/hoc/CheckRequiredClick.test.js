import { createShallow } from '@material-ui/core/test-utils';
import React from 'react';
import { CheckRequiredClick } from './CheckRequiredClick';
import PrStatusActionButton from '../pr/prDetail/PrStatusActionButton';

describe('CheckRequiredClick Component', () => {
  let shallow = createShallow({ dive: false });

  it('should match Snapshot', () => {
    let component = shallow(
      <CheckRequiredClick
        onClick={jest.fn()}
        check={jest.fn()}
        message={'Fehlermeldung.'}
      >
        <PrStatusActionButton label={'Freigabe'} />
      </CheckRequiredClick>
    );

    expect(component).toMatchSnapshot();
  });
});
