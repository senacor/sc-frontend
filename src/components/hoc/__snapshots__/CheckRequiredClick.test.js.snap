// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckRequiredClick Component should match Snapshot 1`] = `
<div>
  <WithStyles(PrStatusActionButton)
    label="Freigabe"
    releaseButtonClick={[Function]}
  />
  <WithStyles(SimpleErrorDialog)
    message="Fehlermeldung."
    onClose={[Function]}
    open={false}
  />
</div>
`;
